src/extras/primitives/registerPrimitive.js

```diff
@@ -75,6 +69,33 @@ module.exports = function registerPrimitive (name, definition) {
         }
       },
 
+      applyDefaultComponents: {
+        value: function () {
+          var self = this;
+          var defaultData = cloneObject(this.defaultAttributes);
```

Comment:
I wasn't clear from the `cloneObject` docstring - why don't we just use `utils.extend` (which is `Object.assign` under the hood - and polyfilled if not natively available)? `Object.assign` does shallow object cloning, one level only, which seems to be what you want. I'm probably missing something though.